src/main/java/org/jboss/ejb/client/EJBRootContext.java

```diff
@@ -48,15 +49,23 @@
     EJBRootContext(final NamingProvider namingProvider, final FastHashtable<String, Object> env) {
         super(env);
         this.namingProvider = namingProvider;
-        final URI providerUri = namingProvider == null ? null : namingProvider.getProviderUri();
-        if (providerUri == null) {
-            affinity = Affinity.NONE;
+
+        // check if strong affinity for this proxy has been set in the environment
+        URI clusterAffinity = getClusterAffinityURIFromEnvironment();
+        if (clusterAffinity != null) {
+            affinity = Affinity.forUri(clusterAffinity);
```

Comment:
...this should accept a cluster name, not a URI, and use the (public) ClusterAffinity(String) constructor.